src/EditorFeatures/Core.Wpf/InlineParamNameHints/InlineParamNameHintsTagger.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Formatting;
+using Microsoft.VisualStudio.Text.Tagging;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.InlineParamNameHints
+{
+    class InlineParamNameHintsTagger : ITagger<IntraTextAdornmentTag>
+    {
+        private readonly ITagAggregator<InlineParamNameHintDataTag> _tagAggregator;
+        private readonly ITextBuffer _buffer;
+
+        public InlineParamNameHintsTagger(ITextBuffer buffer, ITagAggregator<InlineParamNameHintDataTag> tagAggregator)
+        {
+            _buffer = buffer;
+            _tagAggregator = tagAggregator;
+            _tagAggregator.TagsChanged += _tagAggregator_TagsChanged;
+        }
+
+        private void _tagAggregator_TagsChanged(object sender, TagsChangedEventArgs e)
+        {
+            var spans = e.Span.GetSpans(_buffer);
+            foreach (var span in spans)
+            {
+                TagsChanged?.Invoke(this, new SnapshotSpanEventArgs(new SnapshotSpan(span.Snapshot, 0, span.Snapshot.Length)));
```

Comment:
Oops did we need to return this to the original code?

Reply:
I tested it yesterday and thought I found a bug, but haven't been able to reproduce it. I think the original code should be fine but I need to see if the bug comes up.

Reply:
we appear to be refreashing teh entire document on every reported tag.  that's bad.

Reply:
Changed it back to its original state